File: 1963-minimum-number-of-swaps-to-make-the-string-balanced/1963-minimum-number-of-swaps-to-make-the-string-balanced.cpp
class Solution {
public:
    int minSwaps(string s) {
        int c=0,k=0;
        for(auto ch:s)
        {
            if(ch=='[')
                c++;
            else
                c--;
            if(c<0){
                k+=1;
                c=0;
            }

        } 
        return (k+1)/2;
        
    }
};

File: 1963-minimum-number-of-swaps-to-make-the-string-balanced/1963-minimum-number-of-swaps-to-make-the-string-balanced.java
class Solution {
    public int minSwaps(String s) {
        Stack<Character>st = new Stack<>();
        for(int i=0;i<s.length();i++){
            if(s.charAt(i)==']'){
                if(!st.empty()){
                    if(st.peek()=='[')
                        st.pop();
                }
                else
                    st.push(s.charAt(i));
            }
            else
                st.push(s.charAt(i));
        }
        return st.size()/2;
    }
}